rs/nervous_system/tools/sync-with-released-nervous-system-wasms/src/main.rs

```diff
@@ -52,6 +69,139 @@ async fn get_mainnet_canister_git_commit_id_and_module_hash(
     Ok((git_commit_id, module_hash))
 }
 
+/// This function fetches the given canister's module hash deployed on the ICP mainnet (based on `canister_id`)
+/// and then finds a git tag in the given `canister_repository` (e.g., `dfinity/cycles-ledger`)
+/// which contains a release asset whose name matches `canister_name`
+/// and whose sha256 hash matches the module hash deployed on the ICP mainnet
+/// (the sha256 hash is fetched from a dedicated release asset whose name has the form `{canister_name}.sha256`).
+/// To find the git tag, the function proceeds as follows:
+///   - it crawls all git tags of the given `canister_repository`;
+///   - for every git tag, it lists all release assets and looks for a release asset whose name has the form `{canister_name}.sha256`:
+///     if that release asset contains the expected module hash (as deployed on the ICP mainnet),
+///     then the corresponding tag is returned.
```

Comment:
Idea for future work: canisters should store the URL where to retrieve their WASM in their metadata. That way this function can be simplified to just retrieve that URL from the metadata, download it, verify if its hash matches the module hash and if so store that URL in `mainnet-canister-revisions.json`. That way all this interfacing with GitHub would become unnecessary.

Reply:
Good point! https://dfinity.atlassian.net/browse/NNS1-3987

This would require changes in other teams' flows (publishing their WASMs), too, so I'd expect this improvement to not be available soon.